crates/primitives/src/transaction/mod.rs

```diff
@@ -715,6 +715,7 @@ impl reth_codecs::Compact for Transaction {
             }
             #[cfg(feature = "optimism")]
             Self::Deposit(tx) => {
+                let tx: reth_optimism_codecs::TxDepositEncode<'_> = tx.into();
```

Comment:
why is this necessary, seems odd?

Reply:
intermediary type needed cause can't impl `Compact` on op-alloy type in crate where `Compact` is not defined